src/MgvOfferMaking.sol

```diff
@@ -171,7 +171,7 @@ contract MgvOfferMaking is MgvHasOffers {
         provision = 10 ** 9 * offerDetail.gasprice() //gasprice is 0 if offer was deprovisioned
           * (offerDetail.gasreq() + offerDetail.offer_gasbase());
         // credit `balanceOf` and log transfer
-        creditWei(msg.sender, provision);
+        creditWei(msg.sender, provision, int(offerId));
```

Comment:
Should the offerId be negative here since the offer is being deprovisioned?

Reply:
Yes, you might be right, that it would be cleaner to do that here. But it is not needed, as the amount we credit here, is the actual amount that the offer had locked. Where as in the "applyPenalty" scenario, it is the difference of what the taker got minus what  provision was on the offer, that gets credited to the maker. Meaning that by just reading the amount credited in the "applyPenalty" scenario, is not be sufficient to keep a balance of provision of the offer.

An example might be easier to understand:
We post and offer and fund mangrove with 10 matic, (but the offer only needs 1), then these credit/debit event happens.
credit 10 matic to maker. (no offerId known here)
debit 1 matic to maker, with the offerid (lets say offerId = 1)

The offer then gets taken, but fails. We calculate the penalty to 0.5 matic. Then this event happens
credit (1-0.5=0.5) matic to maker with offer id 1.

In this scenario it now looks like the offer still has 0.5 provision locked, but that amount has actually been send as bounty to the taker and the offer is fully deprovisioned.

If we take the same scenario, but the user retracts its own offer and deprovisions it:
Same 2 first events:
credit 10 matic to maker. (no offerId, known here)
debit 1 matic to maker, with the offerid (lets say offerId = 1)

The user now retracts the offer and deprovisions it:
credit 1 matic to maker, with offerId 1

The provision balance of the offer is now 0 and we can therefor see that it is fully deprovisioned.

Because of this, it is only necessary to have a negative offerId to indicate deprovision in the first case. But i would not be opposed to also have it negative in the second case.

Reply:
I don't feel strongly on how we solve this, I just think it's important that the meaning of the negative offer ID is clear and used consistently.

The current description of the event says:
> If the offerId is negative, then we are fully deprovisioning the offer */

With that description, the offer ID should be negative here, right?

Perhaps there's a more precise way to describe what a negative ID means, so that it does not have to be negative here?